### ISSUE TITLE
Clarify roadmap process for the community

### ISSUE THREAD

**Issue by jbw976** [crossplane/crossplane#2482]
### What problem are you facing?

I've seen more users recently wondering about the [project ROADMAP](https://github.com/crossplane/crossplane/blob/master/ROADMAP.md) and noticing that it has gone stale.  This is causing some confusion and lack of clarity for the community that want to know what to expect in the project going forward.  There's a lot to be excited about in the future of Crossplane 💪 

### How could Crossplane help solve your problem?

Some ideas for things we can do to address:

* Remove the ROADMAP document entirely, or keep it up to date
* Point folks to the Projects feature of Github which we are keeping up to date with important issues and ongoing work: https://github.com/orgs/crossplane/projects.  We basically only do this 1 release at a time, which can be OK if we are clear about it.
* Clarify for folks about how they can voice support for issues that are important to them (thumbs-up reaction on the issue, adding comments, attending the community meeting, etc.)

**Comment by negz** [comment]
I agree with @jbw976's suggestions here. If we do want to maintain a roadmap that extends more than one version out I'd personally prefer something like GitHub's public roadmap - https://github.com/github/roadmap/projects/1. This is the approach we use internally at @upbound, and I find it both easier to read and (presumably) easier to update than a markdown file.

Another interesting factor to consider if we were to return to maintaining a longer term roadmap is its scope. Would it contain only "core" Crossplane (i.e. this repo) or include providers as well?
